crates/router/src/core/payment_methods/cards.rs

```diff
@@ -526,9 +602,16 @@ async fn filter_payment_country_based(
 ) -> errors::CustomResult<bool, errors::ApiErrorResponse> {
     Ok(address.map_or(true, |address| {
         address.country.as_ref().map_or(true, |country| {
-            pm.accepted_countries
-                .clone()
-                .map_or(true, |ac| ac.contains(country))
+            pm.accepted_countries.clone().map_or(true, |ac| {
```

Comment:
You can use `as_ref()` instead of `clone()` here.

Reply:
yes